index.js

```diff
@@ -2984,37 +2984,55 @@
 
   //. ### Array
 
-  //# append :: a -> Array a -> Array a
+  //# append :: (Applicative f, Semigroup (f a)) => a -> f a -> f a
```

Comment:
That's a very helpful explanation, @cust0dian! The third type needs parens, though. ;)